valhalla/sif/dynamiccost.h

```diff
@@ -34,17 +34,26 @@
  * @param costing_options  pointer to protobuf costing options object
  * @param range            ranged_default_t object which will check any provided values are in range
  * @param json             rapidjson value object which should contain user provided costing options
- * @param json_key         the json key to use to pull a user provided value out of the jsonn
+ * @param json_key         the json key to use to pull a user provided value out of the json
  * @param option_name      the name of the option will be set on the costing options object
  */
 
-#define JSON_PBF_RANGED_DEFAULT(costing_options, range, json, json_key, option_name)                 \
+#define JSON_PBF_RANGED_DEFAULT(costing_options, range, json, json_key, option_name, warnings)       \
   {                                                                                                  \
+    bool clamped = false;                                                                            \
     costing_options->set_##option_name(                                                              \
         range(rapidjson::get<decltype(range.def)>(json, json_key,                                    \
                                                   costing_options->has_##option_name##_case()        \
                                                       ? costing_options->option_name()               \
-                                                      : range.def)));                                \
+                                                      : range.def),                                  \
+              clamped));                                                                             \
+    if (clamped) {                                                                                   \
+      auto warning = warnings.Add();                                                                 \
+      warning->set_description("'" + std::string(json_key) + "' has been clamped to " +              \
+                               std::to_string(range.def));                                           \
+                                                                                                     \
```

Comment:
try to call `add_warning`

Reply:
@kevinkreiser if we'd like to set this warning status inside the macros, we'd have to
- include the whole `worker.h` into every costing model (which is quite a bit extra to compile?)
- `add_warning` expects the full request, which means I'd have to refactor everything again and also touch the `ParseCosting` etc to not accept the warnings but the full request

so it'd even lead a bigger refactor than what it is currently, unless we're ok that functions accept a `CostingOptions` pbf on top of the full request (where `CostingOptions` comes from). 

are you ok if we just leave it like this and set the warning directly in the macros, i.e. leave the PR as it is?